### ISSUE TITLE
Visual mode: request: make VIM keybindings available in visual mode

### ISSUE THREAD

**Issue by rasmusrhl** [rstudio/rstudio#7973]
### Feature request: Add VIM to visual mode
The new visual mode is awesome. Unfortunately VIM keybindings are not available here. 

I always use VIM keybindings when editing *.r and *.Rmd files. Adding VIM keybindings to visual mode would make this featur even better. 

Thank you for Rstudio :)

**Comment by ronblum** [comment]
@rasmusrhl Thank you for the understandable request! I don't know if this is by design or if there's a way to do so, so I'll mark this as an enhancement to review as we continue to develop RStudio. And thank you for the kind words!

**Comment by mikebessuille** [comment]
@rasmusrhl Do you expect VIM key bindings in the entire visual editor, or just inside R code chunks?

**Comment by davidski** [comment]
> 
> 
> @rasmusrhl Do you expect VIM key bindings in the entire visual editor, or just inside R code chunks?

I'm not the OP, but I expect VIM compatibility throughout the editor. More generally, I very much want vim-mode to improve throughout the platform. VSCode is really spoiling me for an editor experience these days. 😉

**Comment by rasmusrhl** [comment]
> @rasmusrhl Do you expect VIM key bindings in the entire visual editor, or just inside R code chunks?

I agree with @davidski. Ideally VIM key bindings are available in the entire visual editor, not just inside the R code chunks :)

**Comment by CoryMcCartan** [comment]
Just to offer another perspective, I'd be fine with VIM mode only inside code chunks, where the font is monospaced and movement/line copying/etc. commands are most useful.  
This would be a really nice feature to have, and is really the only thing holding me back from defaulting to visual mode.

**Comment by davidzgli** [comment]
I love to have VIM binding too.  Actually, it would be great to have full supports for VIM, not just some keybindings. Yes, vscode has such as supports.  As far as I know, it should not be hard to achieve via neovim.

**Comment by jmcphers** [comment]
I did some exploratory work on this branch: https://github.com/rstudio/rstudio/tree/feature/visual-chunk-keybindings

It's mostly not too hard but there are two major issues:

- Ace's Vim mode (copied from CodeMirror) has a pretty hard/low-level override for movement keys. This means that when you're in Command Mode, our own ProseMirror overrides don't work, so the cursor gets stuck in the editor. We'd have to figure out how to override or monkey-patch Vim mode's handlers, or make changes on our fork. 
- The Vim command line (i.e. the prompt line that appears when you type `:`) doesn't show up in the right place; it's _kind of_ at the bottom of the chunk editor, but since the bottom of the chunk editor may well be offscreen that's a problem. It is also not something we can hook or reposition easily since it, too, is an internal Vim mode component.

**Comment by davidzgli** [comment]
@jmcphers I have no idea how Ace's VIM mode is achieved in Rstudio.  But maybe neovim is a once-and-for-all solution, as VScode neovime (https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim).
